### PR TITLE
ToDo's WPF

### PR DESCRIPTION
**Pull request by brunoss**
Yeah just the simple ToDo's control implemented in WPF. If there is something amiss let me know.

### CODE REVIEW

**Comment by brunoss** [comment]
@retailcoder For the tree view do you have any kind of leveling in mind? I thought of two possibilites:

 - You have a todo per line
 - you have a project, with all modules has children and the todos respectivly

**Comment by Hosch250** [comment]
We discussed this in chat.  I think the consensus is to group like this, which an option to sort the items by type and line number:

Project1
    Module1
        Item1
        Item2
    Module2
        Item1
Project2
    ...

Please see the discussion starting here: http://chat.stackexchange.com/transcript/14929?m=27009422#27009422

When you sort by type, it should order by type, then line number, like:
```
Bug - line 1
Bug - line 2
```
Not:
```
Bug - line 2
Bug - line 1
```

**Comment by brunoss** [comment]
@Hosch250 Seems good to me

**Comment by ckuhn203** [comment]
Just playing Devil's advocate here... Visual Studio doesn't use a tree for this. They use a grid. Just saying.

**Comment by brunoss** [comment]
I think the tree layout will fit nicely on the layout even if it's not how VS do it's To do list.

**Comment by ckuhn203** [comment]
Fair enough. Just pointing out that we might be making a big ado about nothing.

**Comment by brunoss** [comment]
Seems that in the end we're going with a grouping grid, grouped by type. And it also seems that in may be a reused control on other windows.

**Comment by brunoss** [comment]
This was a very direct approach, I guess I'll be getting many criticism about the way I designed the generic control. Such as 'your items are always expanded' or 'The grouping columns could be a bindable property' or even 'You have to use a ListCollectionView which doesn't allow the same functionality as the ObservableCollection'

**Comment by brunoss** [comment]
Tackled the IsExpanded problem which is the only one that has impact on the user side. If the other things comes a bother later I can come back to it. Any other problems?

File: RetailCoder.VBE/ToDoItems/ToDoItem.cs [hunk 1]
```diff
@@ -27,6 +28,9 @@ public class ToDoItem
         private readonly int _lineNumber;
         public int LineNumber { get { return _lineNumber; } }
```
Comment (retailcoder) [inline RetailCoder.VBE/ToDoItems/ToDoItem.cs]: Isn't the `LineNumber` redundant, given the `QualifiedSelection`? It shortens the binding string I guess, does it?

File: RetailCoder.VBE/UI/ToDoItems/ToDoExplorerDockablePresenter.cs [hunk 2]
```diff
@@ -18,111 +18,10 @@ namespace Rubberduck.UI.ToDoItems
...
-            var component = firstOrDefault.VBComponents.Cast<VBComponent>()
-                .First(c => c.Name == e.SelectedItem.ModuleName);
-
-            if (component == null)
-            {
-                return;
-            }
-
-            var codePane = _wrapperFactory.Create(component.CodeModule.CodePane);
-            codePane.Selection = e.SelectedItem.GetSelection().Selection;
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/ToDoItems/ToDoExplorerDockablePresenter.cs]: I love seeing logic moved out of these presenters :)

**Comment by retailcoder** [comment]
Ready to merge?

File: RetailCoder.VBE/ToDoItems/ToDoItem.cs (hunk 1 above)
Comment (brunoss) [inline RetailCoder.VBE/ToDoItems/ToDoItem.cs]: I didn't quite guet it. What you could have said is that `GetSelection` should be turned into a property and `LineNumber` would be binded as `Selection.StartLine`

**Comment by ckuhn203** [comment]
Hey @brunoss, thanks for the PR! We appreciate it!

**Pull request merged**